### DISCUSSION TITLE
Custom background in arbitrary values with Vite doesn't work after build

### DISCUSSION THREAD

**Discussion by nekooee** [tailwindlabs/tailwindcss#9756] [Help] [answered]
When I'm in dev mode on Vite, the custom background in fly works fine. but when I build it with Vite, it doesn't work And no class is added.

```html
<div class="bg-[url('{{Vite::asset('resources/assets/img/bg.svg')}}')] 
	dark:bg-[url('{{Vite::asset('resources/assets/img/bg-dark.svg')}}')]"
></div>
```
tailwindcss version : 3.2.2
vite version: 3.2.2
vite-plugin-dynamic-import version : 1.2.3
laravel-vite-plugin version : 0.7.0

please help me.
thank you

**Comment by wongjn** [comment, marked as answer]
I'm not sure that this should work at all! Regardless, you could try using CSS variables:
```blade
<div
  class="bg-[image:var(--bg)] dark:bg-[image:var(--dark-bg)]"
  style="
    --bg:url('{{ Vite::asset('resources/assets/img/bg.svg') }}');
    --dark-bg:url('{{ Vite::asset('resources/assets/img/bg-dark.svg') }}');
  "
><div>
```

**Reply by nekooee** [reply]
> I'm not sure that this should work at all! Regardless, you could try using CSS variables:
> 
> ```
> <div
>   class="bg-[image:var(--bg)] dark:bg-[image:var(--dark-bg)]"
>   style="
>     --bg:url('{{ Vite::asset('resources/assets/img/bg.svg') }}');
>     --dark-bg:url('{{ Vite::asset('resources/assets/img/bg-dark.svg') }}');
>   "
> ><div>
> ```
Thank you
Yes, it works correctly. But why doesn't it work directly, but it works in a variable way?

**Reply by wongjn** [reply]
Tailwind needs to see the class names as unbroken, full strings, not strings constructed via interpolation or concatenation as you have done here. This is because Tailwind does not run your code and does not know the possibilities of what the code might render - it only sees the files as literal strings (think like if the file ended in `.txt` instead of `.blade.php`). Read more about this in [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).
